### DISCUSSION TITLE
Where are the default OhMyZSH aliases located?

### DISCUSSION THREAD

**Discussion by TheAwesomeGuy47** [ohmyzsh/ohmyzsh#10591] [Help] [answered]
Hi,

I want to replace the default OhMyZSH aliases (found by running `alias`) with my own. However, I am struggling to find where the default aliases are located. Does anybody know where they are?

Thanks!

EDIT: The ones that are there when you first start oh-my-zsh + git plugin aliases

**Comment by mcornella** [comment]
- git-related aliases are defined in [the git plugin](https://github.com/ohmyzsh/ohmyzsh/blob/master/plugins/git/git.plugin.zsh)

- the rest of the aliases defined by default are the following:
  - directory-related: [`lib/directories.zsh`](https://github.com/ohmyzsh/ohmyzsh/blob/master/lib/grep.zsh)
  - grep aliases: [`lib/grep.zsh`](https://github.com/ohmyzsh/ohmyzsh/blob/master/lib/grep.zsh)
  - some misc. aliases in [`lib/misc.zsh`](https://github.com/ohmyzsh/ohmyzsh/blob/master/lib/misc.zsh)
  - ls color aliases in [`lib/theme-and-appearance.zsh`](https://github.com/ohmyzsh/ohmyzsh/blob/master/lib/)

**Reply by TheAwesomeGuy47** [reply]
can I do `source ALIAS_FILE` in these to put them elsewhere? e.g .config

**Reply by mcornella** [reply, marked as answer]
What do you mean? If you want to define other aliases and put them in another file, you can use a custom file that you put in `$ZSH_CUSTOM/<name>.zsh` (e.g. `$ZSH_CUSTOM/aliases.zsh`) and that will automatically get loaded by Oh My Zsh.

Another option is to put them in any file you want and then `source` it from your zshrc file (make sure to put the `source` line after OMZ is sourced).

**Comment by ziofil** [comment]
Probably related to what I'm trying to do: I have my own aliases for `ls`, but they get overwritten by omz. How do I ensure that either my aliases are set last, or how do I disable `ls`-related aliases in omz?

EDIT: wait, I just re-read your last message. I had to make sure they are sourced after sourcing OMZ.

**Comment by TheAwesomeGuy47** [comment]
Just got a notification and realised I probably should have closed this ages ago. Really need to work on my memory/setting reminders.

Thanks for the help, people from 2 and ½ years ago!

**Comment by rahul-behera** [comment]
For anyone wondering, this is where the aliases live. 
`~/.oh-my-zsh/plugins/git/git.plugin.zsh`
